LEAF_Request_Portal/sources/Form.php

```diff
@@ -1008,9 +1009,9 @@ public function getRecordInfo($recordID)
             );
         }
 
-        $dir = new VAMC_Directory;
-        $user = $dir->lookupLogin($res[0]['userID']);
-        $name = isset($user[0]) ? "{$user[0]['Fname']} {$user[0]['Lname']}" : $res[0]['userID'];
+        $userMetadata = json_decode($res[0]['userMetadata'], true);
+        $name = isset($userMetadata) && trim("{$userMetadata['firstName']} {$userMetadata['lastName']}") !== "" ?
+            "{$userMetadata['firstName']} {$userMetadata['lastName']}" : $res[0]['userID'];
```

Comment:
{$userMetadata['firstName']} {$userMetadata['lastName']} gets used a lot in this scope.  Both as a variable and as a determinant.  I'm thinking the first time it comes up, two variables should be created:  A string with the value and a binary for if it exists.

Reply:
That doesn't resolve the issue I brought up.  The efficiency wasn't the concern, it was the repeated code.